static/search.js

```diff
@@ -0,0 +1,72 @@
+function debounce(func, timeout = 200) {
+    let timer;
+    return (...args) => {
+        clearTimeout(timer);
+        timer = setTimeout(() => {
+            func.apply(this, args);
+        }, timeout);
+    };
+}
+
+let boxUpper = document.getElementById('searchBoxUpper');
+let boxLower = document.getElementById('searchBoxLower');
+let postElements = document.getElementsByClassName('post-item');
+let list = [];
+
+(async () => {
+    let response = await fetch('./api/v1/meta')
+    list = await response.json()
+    
+    const options = {
+        includeScore: true,
+        threshold: 0.4,
+        keys: ['title', 'author', 'text']
+    }
+    
+    const fuse = new Fuse(list, options)
+    
+    const runSearch = (() => {
+        let result = [];
+        
+        console.log("new")
+        if (boxUpper.value.length > 0) {
+            result = fuse.search(boxUpper.value)
+            for (let element of postElements) {
+                element.style.display = "none"; 
+            }
+
+            for (let item of result) {
+                for (let element of document.getElementsByClassName(`post-url-${item.item.url}`)) {
+                    element.style.display = "";
+                }
+            }
+        } else if (boxLower != null && boxLower.value.length > 0) {
+            result = fuse.search(boxLower.value)
+            for (let element of postElements) {
+                element.style.display = "none"; 
+            }
+
+            for (let item of result) {
+                for (let element of document.getElementsByClassName(`post-url-${item.item.url}`)) {
+                    element.style.display = "";
+                }
+            }
+        } else {
+            for (let element of postElements) {
+                element.style.display = ""; 
+            }
+        }
+    })
+
+    const trySearch = debounce(() => runSearch());
+    
+    boxUpper.addEventListener("keyup",(e) => {
+        trySearch();
+    });
```

Comment:
don't you have to check for the existence of `boxUpper` here too? perhaps you can use my above suggestion and just have one variable for the search box so that you don't have to worry about setting multiple listeners

Reply:
I'll use your suggestion. boxUpper actually exists always, but boxLower only exists on the main page (long story).